CMakeLists.txt

```diff
@@ -64,31 +60,31 @@ include(check_packages)
 include(swig)
 include(coveralls)
 
-# add PaddlePaddle version
-if(DEFINED ENV{PADDLE_VERSION})
-    add_definitions(-DPADDLE_VERSION=\"$ENV{PADDLE_VERSION}\")
+# Set PaddlePaddle version to Git tag name or Git commit ID.
+find_package(Git REQUIRED)
+execute_process(
+  COMMAND ${GIT_EXECUTABLE} describe --tags --exact-match --abbrev=0
+  WORKING_DIRECTORY ${PROJ_ROOT}
+  OUTPUT_VARIABLE GIT_TAG_NAME
+  RESULT_VARIABLE GIT_RESULT
+  ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
+if(NOT ${GIT_RESULT})
+  set(PADDLE_VERSION ${GIT_TAG_NAME})
```

Comment:
There is another thing in Paddle. In Paddle, every version should be semantic version style, like `major.minor.patch`, such as `0.9.0` or `0.9.0b0`. Any other style of version will cause **ERRORS** or **WARNINGS** in our other scripts.

The possible error scripts are:

* Every python `setup.py`.  https://github.com/PaddlePaddle/Paddle/blob/develop/python/setup.py.in#L15
* Every version compare scripts.

Please don't change the version settings here. Or we should only use git tag as version, and make sure the git tag fit the semantic version style. The possible code could be:

```python
tag_name = get_latest_tag_name()
assert tag_name.is_sematic_versioning()
set(PADDLE_VERSION tag_name)
```

But it is no need implement in CMake scripts. Just bumping up version number is the simplest way to manage version.